### PR TITLE
Initialize repo

### PR DESCRIPTION
**Pull request by balsoft**

### CODE REVIEW

File: terraform/mekbuda.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  vpc_security_group_ids = [
+    aws_security_group.egress_all.id,
+    aws_security_group.mtg.id,
+    aws_security_group.prometheus_exporter_node.id,
+    aws_security_group.ssh.id,
+  ]
+
+  # Instance parameters
+  instance_type = "t3a.micro"
```
Comment (mkaito) [inline terraform/mekbuda.tf:17]: nano?

File: terraform/mekbuda.tf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  monitoring = true
+
+  # Disk type, size, and contents
+  lifecycle { ignore_changes = [ ami ] }
+  ami = data.aws_ami.nixos.id
+  root_block_device {
+    volume_type = "gp2"
+    volume_size = "20"
```
Comment (mkaito) [inline terraform/mekbuda.tf:25]: Can we reduce this?

**Review by mkaito — CHANGES_REQUESTED** [review]

**Comment by mkaito** [comment]
Don't forget to add the cluster and server to the list on [the wiki](https://www.notion.so/serokell/Server-Naming-Scheme-c189819000164fb090377c75e4ce7da6)

File: servers/mekbuda/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ modulesPath, inputs, config, lib, ... }:
+{
+  imports = [
+    "${modulesPath}/virtualisation/amazon-image.nix"
```
Comment (zhenyavinogradov) [inline servers/mekbuda/default.nix:4]: I would also add `systemd.services.amazon-init.enable = false`, to be sure that amazon-init script does not run nixos-rebuild on boot

File: terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,161 @@
+terraform {
+  backend "s3" {
+    bucket = "serokell-gemini-tfstate"
+    dynamodb_table = "serokell-gemini-tfstate-lock"
+    encrypt = true
+    key    = "gemini/terraform.tfstate"
+    region = "eu-west-2"
+  }
+  ## Prevent unwanted updates
+  required_version = "~> 0.12.29" # Use nix-shell or nix develop
```
Comment (zhenyavinogradov) [inline terraform/main.tf:10]: `=` instead of `~>` should be better here, so we don't accidentally update the state to 0.12.30 for example

**Review by zhenyavinogradov — APPROVED** [review]

File: terraform/main.tf [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  assign_ipv6_address_on_creation = true
+
+  enable_nat_gateway = true
+
+  enable_dns_hostnames = true
+  enable_dns_support = true
+
+  enable_dhcp_options              = true
+  dhcp_options_domain_name         = "gemini.serokell.team"
+  dhcp_options_domain_name_servers = ["10.0.0.2"]
```
[the remark this answers is not in the document]
Comment (zhenyavinogradov) [reply terraform/main.tf:152]: It's a standard dns server provided by AWS, but we don't have to add a dhcp option for it ourselves, AWS already specifies it by default

**Review by zhenyavinogradov — APPROVED** [review]
We also need a buildkite pipeline for this repo, other than that LGTM

File: .buildkite/pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+steps:
+  - label: Check Nix flake
+    commands:
+      - nix-shell --run 'nix --experimental-features "nix-command flakes" flake check -L'
```
Comment (mkaito) [inline .buildkite/pipeline.yml:4]: `--experimental-features` is reserved to trusted users.

File: flake.nix [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+          # '';
+        }) nixpkgs.legacyPackages;
+    in {
+      nixosConfigurations = mapAttrs (const mkSystem) servers;
+
+      nixosModules = import ./modules;
+
+      overlay = import ./packages;
+
+      packages = nixpkgs.lib.genAttrs [ "x86_64-linux" "aarch64-linux" ]
```
Comment (mkaito) [inline flake.nix:55]: Did we have an `aarch64-linux` builder?

Comment (balsoft) [reply flake.nix:55]: https://github.com/serokell/aquarius-infra/pull/23

**Review by mkaito — APPROVED** [review]

**Pull request merged**